### PR TITLE
Add optional configuration param for BT Smart Hub6 integration (#31292)

### PR DESCRIPTION
**Pull request by leroyshirto**
## Proposed change

Documents a new optional configuration option for the BT smarthub 2 integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
- This PR fixes or closes issue: https://github.com/home-assistant/home-assistant/pull/31292

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/bt_smarthub.markdown
Comment (frenck) [inline source/_integrations/bt_smarthub.markdown]: I think this is very confusing for our users. Is the default 1 or 2?

Comment (leroyshirto) [reply source/_integrations/bt_smarthub.markdown]: The options are null, 1 and 2, with the default being null. This is a power user config option which allows you to set which model of the smart hub you have. The majority of our users should never need to set the `smarthub_model` config option as the component will try and detect the model if no conf is set.

But I appreciate that it is confusing and needs to be improved because I just wrote 3 sentences explaining it to you lol. I'm not sure what our convention is for explaining this kind of optional config.

Comment (frenck) [reply source/_integrations/bt_smarthub.markdown]: Well let's leave the default out of the documentation in that case 😉

File: source/_integrations/bt_smarthub.markdown
Comment (frenck) [inline source/_integrations/bt_smarthub.markdown]:
```suggestion
  description: The model of your BT Smarthub 1 or 2. If this option is omitted, it will attempt to detect the hub model.
```

Comment (leroyshirto) [reply source/_integrations/bt_smarthub.markdown]: Yes that reads much nicer

File: source/_integrations/bt_smarthub.markdown
Comment (frenck) [reply source/_integrations/bt_smarthub.markdown]: Since there is no default, it still should be removed.

```suggestion
```

**Comment by leroyshirto** [comment]
@jxwolstenholme as codeowner of the component are you able to review changes to docs?

**Review by jxwolstenholme — CHANGES_REQUESTED** [review]
I think that it would be good to be a little bit more specific in the description of smarthub_model. Can you say that the option is either 1 or 2 as there are multiple types of smarthub 1 - the python package supports the smarthub 1 type A but not the smarthub 1 type B. Might be good to also put a little note in the docs to that effect as well.

**Review by jxwolstenholme — DISMISSED** [review]
Approving for now, will make reference to the 1A/1B issue separately.

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by frenck** [comment]
Closing PR, since the parent PR has been closed.

**Comment by frenck** [comment]
Parent PR re-opened

**Review by frenck — COMMENTED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Review by jxwolstenholme — APPROVED** [review]
All good

**Pull request merged**
